File: dart-language/charge-system-demo/client-ng-dart/lib/src/dashboard/dashboard_component.dart
import 'package:angular/angular.dart';
import 'package:angular_router/angular_router.dart';
import '../user_service.dart';
import '../route_paths.dart';

@Component(
  selector: 'my-dashbord',
  styleUrls: ['dashboard_component.css'],
  templateUrl: 'dashboard_component.html',
  directives: [coreDirectives],
)
class DashboardComponent implements OnInit {
  final UserService userService;
  String username = '';
  final Router _router;

  DashboardComponent(this.userService, this._router);

 void ngOnInit() async {
   getCurrentUser();
 }
 void getCurrentUser() {
   username = userService.getCurrentUser();
 }
 void navCharge() {
    _router.navigate(RoutePaths.charge.toUrl());
 }
 void navHistory() {
    _router.navigate(RoutePaths.history.toUrl());
 }
 void logout() {
   final res  = userService.logout();
   if (res) {
    _router.navigate(RoutePaths.login.toUrl());
   }
 }
}

File: dart-language/charge-system-demo/client-ng-ts/projects/at-antd/src/app/dashboard/dashboard.component.ts
import { Component, OnInit } from '@angular/core';
import { UserService } from '../user.service';
import { Router } from '@angular/router';

@Component({
  selector: 'ant-dashboard',
  templateUrl: './dashboard.component.html',
  styleUrls: ['./dashboard.component.less']
})
export class DashboardComponent implements OnInit {
  username: string = '';

  constructor(
    private userService: UserService,
    private router:Router
  ) { }

  ngOnInit() {
    this.username = this.userService.getCurrentUser();
  }
  navHistory() {
    this.router.navigateByUrl('/history');
  }
  navCharge() {
    this.router.navigateByUrl('/charge');
  }
  logOut() {
    this.userService.logout();
    this.router.navigateByUrl('/login', { replaceUrl: true });
  }
}
